applications/grafana-metrics/chart/templates/secret-datasources.yaml

```diff
@@ -0,0 +1,25 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ template "grafana.fullname" . }}-datasources
+  labels:
+    grafana_datasource: {{ template "grafana.fullname" . }}
+    {{- include "grafana.labels" . | nindent 4 }}
+type: Opaque
+stringData:
+  datasources.yaml: |-
+    apiVersion: 1
+    datasources:
+      - name: Prometheus-Thanos
+        access: proxy
+        editable: true
+        isDefault: true
+        jsonData:
+          httpHeaderName1: 'Authorization'
+          timeInterval: 5s
+          tlsSkipVerify: true
+        secureJsonData:
+          httpHeaderValue1: 'Bearer $__file{/run/secrets/kubernetes.io/serviceaccount/token}'
```

Comment:
Out of curiosity, how does this "$__file{}" work???

Reply:
It's a provider to read passwords or secrets from files. Magic!! https://grafana.com/docs/grafana/latest/administration/configuration/#file-provider